File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxprofit = 0;
        int mini = INT_MAX;
        
        for(auto it : prices){
            mini = min(it,mini);
            maxprofit = max(maxprofit,it-mini);
        }
        
        return maxprofit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    
    let maxprofit = 0;
    let mini = 1e9;
    
    for(let i =0;i<prices.length;i++){
        mini = Math.min(mini,prices[i]);
        maxprofit = Math.max(maxprofit, prices[i]-mini);
    }
    return maxprofit;
};